internal/errors/errors.go

```diff
@@ -1,4 +1,4 @@
-package backend
+package errors
```

Comment:
I'm not such a fan of this package. Firstly, because it shares its name with the stdlib package `errors`. But secondly, packages should really define their own errors (unless it's a common system-wide error, such as `io.EOF`). In our case, i don't think these errors are so ubiquitous that they deserve their own package.

There's a clear split in error usage:
```
$ rg "errors\.Err"
tests/state_test.go
76:			c.Sx(fmt.Sprintf("%d NO %v", i, errors.ErrSessionNotSelected))

internal/backend/backend.go
89:		return errors.ErrNoSuchUser
132:		return errors.ErrNoSuchUser
162:	return "", errors.ErrNoSuchUser

internal/state/mailbox.go
163:		return nil, errors.ErrNoSuchMailbox
206:		return nil, errors.ErrNoSuchMailbox

internal/state/snapshot.go
55:		return 0, errors.ErrNoSuchMessage
64:		return 0, errors.ErrNoSuchMessage
73:		return nil, errors.ErrNoSuchMessage
82:		return errors.ErrNoSuchMessage
234:		return errors.ErrNoSuchMessage
242:		return errors.ErrNoSuchMailbox
252:		return errors.ErrNoSuchMessage
279:		return 0, errors.ErrNoSuchMessage
293:		return 0, errors.ErrNoSuchMessage
302:		return 0, errors.ErrNoSuchMessage

internal/state/state.go
79:		return errors.ErrNoSuchMailbox
112:		return errors.ErrNoSuchMailbox
146:			return nil, errors.ErrExistingMailbox
183:		return errors.ErrNoSuchMailbox
200:			return Result{}, errors.ErrNoSuchMailbox
206:			return Result{}, errors.ErrExistingMailbox
215:					return Result{}, errors.ErrExistingMailbox
260:				return errors.ErrNoSuchMailbox
279:		return errors.ErrNoSuchMailbox
281:		return errors.ErrAlreadySubscribed
294:		return errors.ErrNoSuchMailbox
296:		return errors.ErrAlreadyUnsubscribed
320:		return errors.ErrNoSuchMailbox
339:		return errors.ErrSessionNotSelected
347:		return errors.ErrNoSuchMailbox
```

I suggest we move these errors to `internal/state/errors.go`:
- `ErrNoSuchMessage`
- `ErrNoSuchMailbox`
- `ErrExistingMailbox`
- `ErrAlreadySubscribed`
- `ErrAlreadyUnsubscribed`
- `ErrSessionNotSelected` (which we could rename to `ErrMailboxNotSelected`)

This can go back to `internal/backend/errors.go`:
- `ErrNoSuchUser`

These two we can delete (they're not used):
- `ErrNoSuchSnapshot`
- `ErrNotImplemented`

Reply:
Noted. Fixed.